x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/CountAggregatorFunction.java

```diff
@@ -78,23 +78,38 @@ public int intermediateBlockCount() {
     }
 
     private int blockIndex() {
-        return countAll ? 0 : channels.get(0);
+        return countAll ? -1 : channels.get(0);
```

Comment:
Technically this was wrong: `countAll` does not rely on blocks, it only relies on page.positionCount